Karma-Khameleons/bangazon-website
# tests/template/model for product categories

## Description
Add description about what this pull request covers.
This creates the model, template, and view for the product categories (HOME) page. Tests are included for the view and the model. 

## Number of Fixes
List of fixes included in this PR.
No fixes, all new code. 

## Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
#10 Customers should be able to filter products by category

## Expected Behavior
Describe the expected behavior
All test should still pass. The template for product category should display with the labels for the product types currently in the database. 

## Steps to Test Solution
List steps to test your solution
Git fetch
Delete migrations folder, if existing
`python manage.py makemigrations bang_app`
`python manage.py migrate`
`python manage.py test bang_app`
Once tests pass (except one on products), run the server:
`python manage.py runserver`
Navigate to the local host at `localhost:8000/products`
The template should display.

## Unit Testing
- [ X] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
- [X ] I certify that all existing tests pass

## Documentation

- [X ] There is new documentation in this pull request that must be reviewed..
- [X ] I added documentation for any new classes/methods
